### ISSUE TITLE
Use consistent notation for 1-D, 2-D, 3-D

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#7149]
Modules usage messages and documentation mixes up the notation for these dimensions.  I have found 2-D, 2D, 2-d, for instance in our materials, and the same online, so there is no one correct answer here.  Suggest we pick one format and stick with it.  Some ideas here.  Maybe @WalterHFSmith can cut through the fog for us?  I can then update it everywhere.

**Comment by WalterHFSmith** [comment]
You want me to recommend which of "2-D, 2D, 2-d" we use to abbreviate "two-dimensional"? I have no idea what is right here and I will be happy with anyone. Some style manuals would recommend hyphenating words when they form a compound adjective, e.g. "the two-dimensional problem is one which has two dimensions". If we want to follow that, then a hyphen between the digit and the letter makes sense to me. As to whether the letter should be lower or upper case, I am agnostic, but I tend to prefer upper case for abbreviations and acronyms, and it seems that is what this is. So, if you want my taste (I didn't think I cared until I started writing this) I would vote for "2-D".
